chall06/sadawi.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+import sys, fileinput, re
+
+def get_biggest(shelfs):
+		biggest = 0
+		for i in range(0, len(shelfs) - 1):
+			if int(shelfs[i]) > int(shelfs[biggest]):
+				biggest = i
+		size = shelfs[biggest]
+		del shelfs[biggest]
+		return int(size)
+
+if len(sys.argv) < 2:
+	lines = []
+	shelfs = input().split()
+	for n in shelfs:
+		if not n.isdigit():
+			sys.exit(sys.argv[0] + ": " + "stdin:" + " Can't read file")
+	while True:
+		try:
+			line = input()
+		except EOFError:
+			break
+		lines.append(line)
+	files = []
+	books = 0
+	for line in lines:
+		try:
+			books += int((re.search(r"\d+", line).group()))
+		except:
+			sys.exit("Invalid input")
+
+	shelf_size = 0
+	shelf_amount = 0
+	while (shelf_size < books):
+		shelf_size += get_biggest(shelfs)
+		shelf_amount += 1
+		if not shelfs:
+			sys.exit(sys.argv[0] + ": " + "stdin:" + " Not enough space in the given shelves")
+	print(shelf_amount)
+else:
+	args = sys.argv[1:]
+	for arg in args:
+		lines = []
+		fail = 0
+		if len(args) > 1:
+			print(arg + ":")
+		try:
+			file = open(arg, "r")
+		except:
+			print(sys.argv[0] + ": " + arg + ": Can't read file")
+			continue
+		shelfs = file.readline().split()
```

Comment:
not sure it has exactly the same behavior but you may take a look to the function readlines() 

https://www.tutorialspoint.com/python3/file_readlines.htm